src/plugins/georeferencer/qgstransformsettingsdialog.cpp

```diff
@@ -31,15 +32,54 @@ QgsTransformSettingsDialog::QgsTransformSettingsDialog( const QString &raster, c
   , mCountGCPpoints( countGCPpoints )
 {
   setupUi( this );
-  connect( tbnOutputRaster, &QToolButton::clicked, this, &QgsTransformSettingsDialog::tbnOutputRaster_clicked );
-  connect( tbnMapFile, &QToolButton::clicked, this, &QgsTransformSettingsDialog::tbnMapFile_clicked );
-  connect( tbnReportFile, &QToolButton::clicked, this, &QgsTransformSettingsDialog::tbnReportFile_clicked );
+
+  QgsSettings settings;
+  restoreGeometry( settings.value( QStringLiteral( "/Plugin-GeoReferencer/TransformSettingsWindow/geometry" ) ).toByteArray() );
+
+  mOutputRaster->setStorageMode( QgsFileWidget::SaveFile );
+  if ( output.isEmpty() )
+  {
+    mOutputRaster->setFilePath( generateModifiedRasterFileName( mSourceRasterFile ) );
+  }
+  else
+  {
+    mOutputRaster->setFilePath( output );
+  }
+  mOutputRaster->setFilter( tr( "TIF files" ) + " (*.tif *.tiff *.TIF *.TIFF)" );
+  mOutputRaster->setDialogTitle( tr( "Destination Raster" ) );
+  mOutputRaster->setDefaultRoot( settings.value( QStringLiteral( "UI/lastRasterFileFilterDir" ), QDir::homePath() ).toString() );
+  connect( mOutputRaster, &QgsFileWidget::fileChanged, this, [ = ]
+  {
+    QgsSettings settings;
+    QFileInfo tmplFileInfo( mOutputRaster->filePath() );
+    settings.setValue( QStringLiteral( "UI/lastRasterFileFilterDir" ), tmplFileInfo.absolutePath() );
+  } );
+
+  mPdfMap->setStorageMode( QgsFileWidget::SaveFile );
+  mPdfMap->setFilter( tr( "PDF files" ) + " (*.pdf *.PDF)" );
+  mPdfMap->setDialogTitle( tr( "Save Map File as" ) );
+  mPdfMap->setDefaultRoot( settings.value( QStringLiteral( "/Plugin-GeoReferencer/lastPDFReportDir" ), QDir::homePath() ).toString() );
+  connect( mPdfMap, &QgsFileWidget::fileChanged, this, [ = ]
```

Comment:
I think handling this should become an option built into QgsFileWidget - you could copy @NathanW2 's recent work with the dialog geometry restore to auto-build the key as required.